File: app/src/main/java/com/riders/thelab/data/remote/dto/ArtistsResponse.java
package com.riders.thelab.data.remote.dto;

import com.squareup.moshi.FromJson;
import com.squareup.moshi.ToJson;

import java.util.List;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Setter
@Getter
@ToString
public class ArtistsResponse {
    private List<Artist> artists;

    @FromJson
    List<Artist> artistsFromJson(List<Artist> artistsEventJson) {
        return artistsEventJson;
    }

    @ToJson
    String artistsToJson(List<Artist> artists) {
        return artists.toString();
    }
}

File: app/src/main/java/com/riders/thelab/data/remote/dto/artist/ArtistsResponse.kt
package com.riders.thelab.data.remote.dto.artist

import com.squareup.moshi.FromJson
import com.squareup.moshi.ToJson

class ArtistsResponse {
    private val artists: List<Artist>? = null

    @FromJson
    fun artistsFromJson(artistsEventJson: List<Artist>): List<Artist> {
        return artistsEventJson
    }

    @ToJson
    fun artistsToJson(artists: List<Artist?>): String {
        return artists.toString()
    }
}